### ISSUE TITLE
[import/no-unused-modules] Typescript `namespace` usage results in false positive 

### ISSUE THREAD

**Issue by UwaisA** [import-js/eslint-plugin-import#2336]
```typescript
// Foo.ts
export namespace MyNamespace {
  export interface Internal {
    a: string;
  }
}
```
```typescript
// Bar.ts
import { MyNamespace } from 'Foo';

export const internal: MyNamespace.Internal = { a: 'a' };
```

```
/src/Foo.ts
  2:3  error  exported declaration 'Internal' not used within other modules      import/no-unused-modules
```

Both the export of `MyNamepsace` and `Internal` seem to be required for the usage to compile in `Bar.ts`. I think this kind of usage should not result in any errors.

This seems to be in a similar area to https://github.com/import-js/eslint-plugin-import/issues/1964 but with a different rule.

**Comment by ekilah** [comment]
I've run into this as well, just commenting to bump this since it's a couple years old.

**Comment by ljharb** [comment]
Using namespaces at all is super unidiomatic, but i agree we shouldn’t give the wrong answer.
